/*
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package host.boottimeproto;

message HostBootDurationExtra {
  string name = 1;
  uint64 milliseconds = 2;
}

message HostBootDuration {
  uint64 os_user_down = 1;
  uint64 os_kernel_down = 2;
  uint64 bmc_down = 3;
  uint64 bmc = 4;
  uint64 bios = 5;
  uint64 nerf_kernel = 6;
  uint64 nerf_user = 7;
  uint64 os_kernel = 8;
  uint64 os_user = 9;
  uint64 unmeasured = 10;
  uint64 total = 11;
  repeated HostBootDurationExtra extra = 12;
}

message HostBootStatistic {
  int32 internal_reboot_count = 1;
  bool power_cycle_type = 2;
}

message HostBootTimeInfo {
  HostBootDuration durations = 1;
  HostBootStatistic statistics = 2;
}
